src/World_TEST.cc

```diff
@@ -691,3 +691,133 @@ TEST(DOMWorld, Plugins)
   world.ClearPlugins();
   EXPECT_TRUE(world.Plugins().empty());
 }
+
+/////////////////////////////////////////////////
+TEST(DOMWorld, LoadEarthSurface)
+{
+  sdf::World world;
+
+  std::ostringstream stream;
+  stream
+    << "<?xml version='1.0'?>"
+    << "<sdf version='1.10'>"
+    << "<world name='spherical coordinates'>"
+    << "<spherical_coordinates>"
+    << " <surface_model>EARTH_WGS84</surface_model>"
+    << "  <world_frame_orientation>ENU</world_frame_orientation>"
+    << "  <latitude_deg>-22.9</latitude_deg>"
+    << "  <longitude_deg>-43.2</longitude_deg>"
+    << "  <elevation>0</elevation>"
+    << "  <heading_deg>0</heading_deg>"
+    << "</spherical_coordinates>"
+    << "</world>"
+    << "</sdf>";
+
+  sdf::SDFPtr sdfParsed(new sdf::SDF());
+  sdf::init(sdfParsed);
+  ASSERT_TRUE(sdf::readString(stream.str(), sdfParsed));
+  auto SphericalCoordinatesSDF =
+    sdfParsed->Root()->GetElement("world");
+  ASSERT_NE(nullptr, SphericalCoordinatesSDF);
+
+  auto errors = world.Load(SphericalCoordinatesSDF);
+  EXPECT_EQ(errors.size(), 0);
+}
+
+/////////////////////////////////////////////////
+TEST(DOMWorld, LoadMoonSurface)
+{
+  sdf::World world;
+
+  std::ostringstream stream;
+  stream
+    << "<?xml version='1.0'?>"
+    << "<sdf version='1.10'>"
+    << "<world name='spherical coordinates'>"
+    << "<spherical_coordinates>"
+    << " <surface_model>MOON_SCS</surface_model>"
+    << "  <world_frame_orientation>ENU</world_frame_orientation>"
+    << "  <latitude_deg>-22.9</latitude_deg>"
+    << "  <longitude_deg>-43.2</longitude_deg>"
+    << "  <elevation>0</elevation>"
+    << "  <heading_deg>0</heading_deg>"
+    << "</spherical_coordinates>"
+    << "</world>"
+    << "</sdf>";
+
+  sdf::SDFPtr sdfParsed(new sdf::SDF());
+  sdf::init(sdfParsed);
+  ASSERT_TRUE(sdf::readString(stream.str(), sdfParsed));
+  auto SphericalCoordinatesSDF =
+    sdfParsed->Root()->GetElement("world");
+  ASSERT_NE(nullptr, SphericalCoordinatesSDF);
+
+  auto errors = world.Load(SphericalCoordinatesSDF);
+  EXPECT_EQ(errors.size(), 0);
+}
+
+/////////////////////////////////////////////////
+TEST(DOMWorld, LoadCustomSurface)
+{
+  sdf::World world;
+
+  std::ostringstream stream;
+  stream
+    << "<?xml version='1.0'?>"
+    << "<sdf version='1.10'>"
+    << "<world name='spherical coordinates'>"
+    << "<spherical_coordinates>"
+    << " <surface_model>CUSTOM_SURFACE</surface_model>"
+    << "  <surface_axis_equatorial>11</surface_axis_equatorial>"
```

Comment:
since these are specified in meters, I would use larger values, at least 10,000 to be more realistic

Reply:
Done : https://github.com/gazebosim/sdformat/pull/1050/commits/d877d71d2d2b20c1754f91825b0c1a73b065b17a